File: 백준/Silver/14929. 귀찮아 （SIB）/귀찮아 （SIB）.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        String[] num = br.readLine().split(" ");
        for(int i=0;i<n;i++){
            arr[i] = Integer.parseInt(num[i]);
        }
        long[] prefix_sum = new long[n+1];
        Arrays.fill(prefix_sum,0);

        for(int i=1;i<n+1;i++){
            prefix_sum[i] = prefix_sum[i-1]+arr[i-1];
        }
        long res = 0;
        for(int i=0;i<n;i++){
            res += (long)arr[i]*(prefix_sum[n]-prefix_sum[i+1]);
        }
        System.out.println(res);
    }
}

File: 백준/Silver/14929. 귀찮아 （SIB）/귀찮아 （SIB）.py
import sys
input = sys.stdin.readline
n = int(input())
arr = list(map(int,input().split()))
prefix_sum = [0] * (n + 1)
for i in range(1, n + 1):
    prefix_sum[i] = prefix_sum[i-1] + arr[i-1]

result = 0
for i in range(n):
    result += arr[i] * (prefix_sum[n] - prefix_sum[i+1])

print(result)